src/hotspot/share/gc/shenandoah/c2/shenandoahBarrierSetC2.cpp

```diff
@@ -1043,8 +1043,10 @@ void ShenandoahBarrierSetC2::verify_gc_barriers(Compile* compile, CompilePhase p
 #endif
 
 void ShenandoahBarrierSetC2::maybe_step_over_cmpp_inputs(Node*& in1, Node*& in2) const {
+  // If first input is NULL, then skip the LRB barriers (except native) on the second input
```

Comment:
OK, as we discussed offline, this also admits other barriers... so maybe:
`// If first input is NULL, then step over the barriers (except LRB native) on the second input`